test/extended/testdata/gssapi/fedora/base/gssapi-tests.sh

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/bash
+
+set -e
+set -x
+
+echo "I ran this on `uname -a`"
+
+CLIENT_HAS_GSSAPI=true
+SERVER_HAS_BASIC=true
+users=(user1 user2 user3 user4 user5)
+
+# Client has no GSSAPI and server is GSSAPI only
+# Everything fails
+
+if [[ !CLIENT_HAS_GSSAPI && !SERVER_HAS_BASIC ]]
+then
+    for u in ${users[@]}
```

Comment:
In general, I don't think we want to just loop through the users. Use them intentionally (e.g. get a ticket for user1, get a ticket with fully qualified principal name for user2, log in with basic for user3, expect user 4 to fail, etc)